crates/sallyport/src/guest/call/enarxcall/passthrough.rs

```diff
@@ -101,36 +101,50 @@ impl PassthroughAlloc for MemInfo {
     }
 }
 
-/// Within an address range inside the enclave, ask host to remove pages
-/// from the enclave. Pages must be trimmed before this operation is
-/// applied.
-pub struct RemoveSgxPages {
+/// Notify the host about `mmap()`.
+pub struct MmapHost {
     pub addr: NonNull<c_void>,
     pub length: usize,
+    pub prot: c_int,
 }
 
-impl PassthroughAlloc for RemoveSgxPages {
-    const NUM: Number = Number::RemoveSgxPages;
+impl PassthroughAlloc for MmapHost {
+    const NUM: Number = Number::MmapHost;
 
-    type Argv = Argv<2>;
-    type Ret = ();
+    type Argv = Argv<3>;
+    type Ret = usize;
```

Comment:
```suggestion
    type Ret = ();
```
Don't return errno as the return value. for the host enarxcall handler, you would do `*ret = -errno`

Reply:
fixed